### ISSUE TITLE
Searching for filtering available files on the load dialog seems broken

### ISSUE THREAD

**Issue by SusanBrown** [cwrc/cwrc-git-dialogs#32]
Search seems totally to be broken

## Expected Behaviour
If I enter the name of a file in part I expect the list to be filtered to show only the available files that have strings matching my search string.

## Current Behaviour
There are no matches when there should be:

![image](https://user-images.githubusercontent.com/545952/62745211-ddd13400-ba17-11e9-98fa-42d6f2939991.png)

![image](https://user-images.githubusercontent.com/545952/62745179-c5f9b000-ba17-11e9-8ec4-9fc1554f31c7.png)

**Comment by ilovan** [comment]
Great catch, @SusanBrown . 
@ajmacdonald , dev-cwr-writer is returning an even weirder result:
![image](https://user-images.githubusercontent.com/6600070/62746782-9003fe00-ba0e-11e9-9524-c34bd238f849.png)

**Comment by ajmacdonald** [comment]
@SusanBrown the search is performed on the text of the files within the repos not the names of the repos themselves

**Comment by SusanBrown** [comment]
Could it search names of files, names of repos, and text content?
On Aug 9, 2019, 2:25 PM -0400, Andrew <notifications@github.com>, wrote:

@SusanBrown<https://github.com/SusanBrown> the search is performed on the text of the files within the repos not the names of the repos themselves

**Comment by ajmacdonald** [comment]
@ilovan I've fixed the issue on dev

**Comment by ajmacdonald** [comment]
@SusanBrown yes that's possible but not straightforward since those are two different calls

**Comment by ajmacdonald** [comment]
Searching file names within repos will be tricky. Whereas there's a method in the GitHub API for [searching within file contents](https://developer.github.com/v3/search/#search-code), there is no equivalent for file names.

You can see this reflected in the interface on GitHub itself.
Searching within a repository does not include file name matches: https://github.com/cwrc/cwrc-git-dialogs/search?q=load&unscoped_q=load
You must instead use a separate page to search by file name: https://github.com/cwrc/cwrc-git-dialogs/find/master

A workaround would require fetching the contents of each repo for a user, then searching within those contents. Depending on the number of repos a user owns, this could be a very expensive operation. A compromise might be to have the user select a repo first, then have the ability to search within that particular repo.

**Comment by SusanBrown** [comment]
What you suggest sounds like a very good approach to me.

On Nov 15, 2019, at 3:02 PM, Andrew <notifications@github.com<mailto:notifications@github.com>> wrote:


Searching file names within repos will be tricky. Whereas there's a method in the GitHub API for searching within file contents<https://developer.github.com/v3/search/#search-code>, there is no equivalent for file names.

You can see this reflected in the interface on GitHub itself.
Searching within a repository does not include file name matches: https://github.com/cwrc/cwrc-git-dialogs/search?q=load&unscoped_q=load
You must instead use a separate page to search by file name: https://github.com/cwrc/cwrc-git-dialogs/find/master

A workaround would require fetching the contents of each repo for a user, then searching within those contents. Depending on the number of repos a user owns, this could be a very expensive operation. A compromise might be to have the user select a repo first, then have the ability to search within that particular repo.

**Comment by ajmacdonald** [comment]
Great! @SusanBrown 
I think we might then also want to consider switching to a multi-column layout for the load/browse dialog, similar to the Mac OS Finder.
Column 1: repo search
Column 2: repo search results
Column 3: selected repo contents, including an input for searching within the repo

**Comment by SusanBrown** [comment]
Sounds good. I always stumble on that particular finder view actually and end up switching away from it ;) but if this is the logical way to go, let’s go for it. Do you need help from Mihaela by way of a mockup or anything?

On Nov 15, 2019, at 3:30 PM, Andrew <notifications@github.com<mailto:notifications@github.com>> wrote:


Great! @SusanBrown<https://github.com/SusanBrown>
I think we might then also want to consider switching to a multi-column layout for the load/browse dialog, similar to the Mac OS Finder.
Column 1: repo search
Column 2: repo search results
Column 3: selected repo contents, including an input for searching within the repo

**Comment by ajmacdonald** [comment]
e6dd57a027e42aeff6a62d906cfaed42f01e28e4

**Comment by ilovan** [comment]
working now on dev-cwrc-writer
